cmd/run/run.go

```diff
@@ -680,17 +708,21 @@ func RunServer(ctx context.Context, config *Config) error {
 			}).Handler(mux),
 		}
 
-		go func() {
-			var err error
-			if config.HTTP.TLS.Enabled {
-				if config.HTTP.TLS.CertPath == "" || config.HTTP.TLS.KeyPath == "" {
-					logger.Fatal("'http.tls.cert' and 'http.tls.key' configs must be set")
-				}
-				err = httpServer.ListenAndServeTLS(config.HTTP.TLS.CertPath, config.HTTP.TLS.KeyPath)
-			} else {
-				err = httpServer.ListenAndServe()
+		listener, err := net.Listen("tcp", config.HTTP.Addr)
+		if err != nil {
+			return err
+		}
+
+		if config.HTTP.TLS.Enabled {
+			watcher, err := newCertWatcher(ctx, logger, config.HTTP.TLS.CertPath, config.HTTP.TLS.KeyPath)
+			if err != nil {
+				return err
 			}
-			if err != http.ErrServerClosed {
+			listener = tls.NewListener(listener, &tls.Config{GetCertificate: watcher.GetCertificate})
+		}
+
+		go func() {
+			if err := httpServer.Serve(listener); err != http.ErrServerClosed {
```

Comment:
To be consistent with the grpc `Serve` above.
```suggestion
			if err := httpServer.Serve(listener); err != nil {
			    if !errors.Is(err, http.ErrServerClosed) {
			        logger.Fatal("HTTP server closed with unexpected error", zap.Error(err))
			    }
			}
```

Reply:
Done

Reply:
@jon-whit Sorry I missed this. I have added the error conditional statement.